src/SDK/SmartDetectorsSDK/AlertPresentation/MetricChart.cs

```diff
@@ -0,0 +1,96 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="MetricChart.cs" company="Microsoft Corporation">
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// </copyright>
+//-----------------------------------------------------------------------
+
+namespace Microsoft.Azure.Monitoring.SmartDetectors.AlertPresentation
+{
+    using System;
+    using System.Collections.Generic;
+
+    /// <summary>
+    /// Represents a chart displaying metric data.
+    /// </summary>
+    public class MetricChart
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="MetricChart"/> class.
+        /// </summary>
+        /// <param name="metricName">The name of the metric to display in the chart.</param>
+        /// <param name="timeGrain">The time grain (resolution) of the metric to display.</param>
+        /// <param name="aggregationType">The aggregation type to use when displaying the metric.</param>
+        public MetricChart(
```

Comment:
>MetricChart [](start = 15, length = 11)

Why not initialize everything in the ctor?

Reply:
i'm only initializing mandatory members here, to make life easier for the developer

---
In reply to: [258926689](https://github.com/Azure/SmartDetectorsMonitoringAppliance/pull/52#discussion_r258926689) [](ancestors = 258926689)